orthoinference/src/main/java/org/reactome/orthoinference/EWASInferrer.java

```diff
@@ -218,10 +186,23 @@
 					infEWASInstances.add((GKInstance) infEWASInst);
 				}
 			}
+		}
 		return infEWASInstances;
 	}
 
-    // Creates ReferenceGeneSequence instance based on ENSG identifier mapped to protein. Creates an instance for the primary database and an alternate, if it exists.
+	// Homologous Protein IDs can exist in ${source}_${target}_mapping.txt but the corresponding Gene ID might not exist in ${target}_gene_protein_mapping.txt.
+	// This is different from when we built Orthopairs files using Compara, since the homology mapping file was generated using IDs from the gene-protein file.
+	// This function prevents a Null Exception from killing the entire Reaction's inference, rather than just the EWAS inference.
+	private static boolean checkValidSpeciesProtein(String homologueId) {
+		List<String> ensgIds = ensgMappings.get(homologueId);
```

Comment:
Is it always the case that if a homologueId key returns a null value from the map it means it isn't in the map or do homologue ids get included in the ensg map even if they have no gene ids to map to?

If homologue ids without gene ids aren't in the map as keys, I think you can do this check like this: 

`return ensgMappings.containsKey(homologueId);`

Reply:
Very cool, wasn't aware of that function